src/app/shared/http-services/group-permissions.service.ts

```diff
@@ -5,25 +5,23 @@ import { map } from 'rxjs/operators';
 import { appConfig } from '../helpers/config';
 import { assertSuccess, SimpleActionResponse } from './action-response';
 import * as D from 'io-ts/Decoder';
-import { permissionsDecoder } from 'src/app/modules/item/helpers/item-permissions';
 import { pipe } from 'fp-ts/function';
 import { decodeSnakeCase } from '../operators/decode';
-import { dateDecoder } from '../helpers/decoders';
+import {
+  itemCorePermDecoder,
+  itemEntryFromPermDecoder,
+  itemEntryUntilPermDecoder,
+  itemSessionPermDecoder
+} from '../models/domain/item-permissions';
 
 const groupPermissionsDecoder = pipe(
-  permissionsDecoder,
-  D.intersect(
-    D.struct({
-      canMakeSessionOfficial: D.boolean,
-      canEnterFrom: dateDecoder
-    })
-  )
+  itemCorePermDecoder,
+  D.intersect(itemSessionPermDecoder),
+  D.intersect(itemEntryFromPermDecoder),
 );
 
 const groupPermissionsInfoDecoder = D.struct({
-  granted: pipe(groupPermissionsDecoder, D.intersect(D.struct({
-    canEnterUntil: dateDecoder
-  }))),
+  granted: D.intersect(groupPermissionsDecoder)(itemEntryUntilPermDecoder),
```

Comment:
In the past you made a remark about using `D.intersect` with `pipe`

Reply:
Same here: https://github.com/France-ioi/AlgoreaFrontend/pull/1096/files#diff-627a1ba3e4c45c92a1eea14c11165a322a2f9204452bb22f2992e4ea9498e38eR25

Reply:
With 2, I found them more readable like that than with a pipe. You prefer I use a pipe?

Reply:
I personally don't care